cms/djangoapps/maintenance/views.py

```diff
@@ -211,3 +226,80 @@ def post(self, request):
             exc_info=True
         )
         return self.render_response()
+
+
+class AnnouncementIndexView(ListView, MaintenanceBaseView):
+    """
+    View for viewing the announcements shown on the dashboard, used by the global staff.
+    """
+
+    model = Announcement
+    object_list = Announcement.objects.all()
+    context_object_name = 'announcement_list'
+    paginate_by = 9
+
+    def __init__(self):
+        super(AnnouncementIndexView, self).__init__(MAINTENANCE_VIEWS['announcement_index'])
+
+    @method_decorator(require_global_staff)
+    def get(self, request, *args, **kwargs):
+        if "limit" in request.GET.keys():
+            self.paginate_by = request.GET.get("limit")
+        context = self.get_context_data()
+        context['view'] = MAINTENANCE_VIEWS['announcement_index']
+        return render_to_response(self.template, context)
+
+
+class AnnouncementEditView(UpdateView, MaintenanceBaseView):
+    """
+    View for editing an announcement.
+    """
+    model = Announcement
+    fields = ['content', 'active']
+    success_url = '/maintenance/announcements'
+    template = '/maintenance/_announcement_edit.html'
+
+    @method_decorator(require_global_staff)
+    def dispatch(self, request, *args, **kwargs):
+        return super(AnnouncementEditView, self).dispatch(request, *args, **kwargs)
+
+    @method_decorator(require_global_staff)
+    def get(self, request, *args, **kwargs):
+        super(AnnouncementEditView, self).get(request, *args, **kwargs)
+        context = self.get_context_data()
+        context['action_url'] = reverse('maintenance:announcement_edit', kwargs={'pk': context['announcement'].pk})
+        return render_to_response(self.template, context)
+
+
+class AnnouncementCreateView(CreateView, MaintenanceBaseView):
+    """
+    View for creating an announcement.
+    """
+    model = Announcement
+    fields = ['content']
+    success_url = '/maintenance/announcements'
+    template = '/maintenance/_announcement_edit.html'
```

Comment:
You can greatly simplify these edit views and their templates by using `template_name` here instead.  See e.g. [ApiRequestView](https://github.com/edx/edx-platform/blob/8c66d867044295eac4ded838028e43e8521bfc9f/openedx/core/djangoapps/api_admin/views.py#L28), and its template, [api_access_request_form.html](https://github.com/edx/edx-platform/blob/6d2468d0dcf782a7561ecc61946146d6fae2762f/lms/templates/api_admin/api_access_request_form.html#L19).

To editing with a textarea, use `widget=forms.Textarea` in the custom form field  (cf https://stackoverflow.com/a/430642/4302112).